iam/snippets/snippets.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package snippets
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+	iam "google.golang.org/api/iam/v1"
+)
+
+var client, _ = google.DefaultClient(
+	context.Background(),
+	iam.CloudPlatformScope)
+var iamService, _ = iam.New(client)
+
+// [START iam_view_grantable_roles]
+func viewGrantableRoles(fullResourceName string) {
```

Comment:
Please have `w io.Writer` as the first argument then write to it with `fmt.Fprintln(w, ...)`. This makes it easier to test, doesn't lead to printing things out when running tests, and is the same as writing to an `http.ResponseWriter` in an HTTP handler. Example: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/monitoring/snippets/channel_delete.go#L21

Reply:
It might be easier to test, but it's harder for users to read and copy/paste from the docs. User friendliness is more important than ease of testing.

Reply:
There are lots of options for how to write samples, and this is the best all-around way we've come up with for writing them. Using `fmt.Fprintf` is common and makes this very easy to convert to an HTTP handler.